File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int maxProfit = 0;
        int minPrice = Integer.MAX_VALUE;

        for (int price : prices) {
            // Update minimum price if the current price is lower
            if (price < minPrice) {
                minPrice = price;
            }
            // Calculate profit by subtracting min price from current price
            int profit = price - minPrice;
            // Update max profit if current profit is higher
            if (profit > maxProfit) {
                maxProfit = profit;
            }
        }
        return maxProfit;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.kt
class Solution {
    fun maxProfit(prices: IntArray): Int {
        var mini =  prices[0]
        var maxProfit = 0

        for(i in 1 until prices.size ){
            if(prices[i] < mini){
                mini = prices[i]
            }

            var profit = prices[i] - mini

            if(maxProfit < profit) maxProfit = profit
        }

        return maxProfit

    }
}